File: app/src/main/java/com/llj/architecturedemo/module/MainActivityModule.java
package com.llj.architecturedemo.module;

import android.arch.lifecycle.ViewModelProviders;

import com.llj.architecturedemo.ui.activity.MainActivity;
import com.llj.architecturedemo.view.MainContractView;
import com.llj.architecturedemo.vm.MainContractViewModel;

import dagger.Module;
import dagger.Provides;

/**
 * ArchitectureDemo
 * describe:
 * author liulj
 * date 2018/5/16
 */
@Module
public class MainActivityModule {
    @Provides
    MainContractView provideView(MainActivity activity) {
        return activity;
    }

    @Provides
    MainContractViewModel provideModel(MainActivity activity) {
        return ViewModelProviders.of(activity).get(MainContractViewModel.class);
    }

}

File: app/src/p_app/main/java/com/llj/architecturedemo/module/MainActivityModule.kt
package com.llj.architecturedemo.module

import com.llj.architecturedemo.ui.activity.MainActivity
import com.llj.architecturedemo.view.MainContractView

import dagger.Module
import dagger.Provides

/**
 * ArchitectureDemo
 * describe:
 * author liulj
 * date 2018/5/16
 */
@Module
class MainActivityModule {
    @Provides
    internal fun provideView(activity: MainActivity): MainContractView {
        return activity
    }

    //    @Provides
    //    MainContractViewModel provideModel(MainActivity activity) {
    //        return ViewModelProviders.of(activity).get(MainContractViewModel.class);
    //    }

}
